org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/project/MavenProjectConfigurator.java

```diff
@@ -88,6 +90,12 @@ public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
                 }
                 // Some errors were detected
                 discoverProposals(discoveryRequest, monitor);
+                Map<ILifecycleMappingRequirement, List<IMavenDiscoveryProposal>> proposals = discoveryRequest
+                    .getAllProposals();
+                if(proposals.isEmpty()) {
```

Comment:
I try to prevent excessive inlining as it is then often hard to debug (you have to look at the internal state of the object instead of just a local variable,  also in stacktraces it is hard to guess what cause the error

So at laest for now I'll keep it that way just in case there are issues and we need to debug things.

Reply:
> I try to prevent excessive inlining as it is then often hard to debug (you have to look at the internal state of the object instead of just a local variable, also in stacktraces it is hard to guess what cause the error

You could just step into the method and step return immediately and see what the debugger shows as return type of the method call.

But if prefer to keep it separated using `var` would be an alternative to make that more readable. The actual type of `proposals` is not really necessary to understand the code. Just in case you come along this method again. :)